controller/workitem.go

```diff
@@ -561,6 +563,199 @@ func getVersion(version interface{}) (int, error) {
+	if fieldValueGeneric != nil && len(fieldValueStr) == 1 {
+		switch kind {
+		case workitem.KindUser:
+			cachedValue, ok := (*uuidStringCache)[fieldValueStr[0]]
+			if ok {
+				return ptr.String(cachedValue), nil
+			}
+			userID, err := uuid.FromString(fieldValueStr[0])
+			if err != nil {
+				return nil, errs.Wrapf(err, "failed to convert user type value to string for field key: %s, value %s", fieldKey, fieldValueStr[0])
+			}
+			user, err := app.Identities().Load(ctx, userID)
+			if err != nil {
+				return nil, errs.Wrapf(err, "failed to retrieve user for field key: %s", fieldKey)
+			}
+			(*uuidStringCache)[fieldValueStr[0]] = user.Username
+			return ptr.String(user.Username), nil
+		case workitem.KindIteration:
+			cachedValue, ok := (*uuidStringCache)[fieldValueStr[0]]
+			if ok {
+				return ptr.String(cachedValue), nil
+			}
+			iterationID, err := uuid.FromString(fieldValueStr[0])
+			if err != nil {
+				return nil, errs.Wrapf(err, "failed to convert iteration type value to string for field key: %s", fieldKey)
+			}
+			iteration, err := app.Iterations().Load(ctx, iterationID)
+			if err != nil {
+				return nil, errs.Wrapf(err, "failed to retrieve iteration for field key: %s", fieldKey)
+			}
+			(*uuidStringCache)[fieldValueStr[0]] = iteration.Name
+			return ptr.String(iteration.Name), nil
+		case workitem.KindArea:
+			cachedValue, ok := (*uuidStringCache)[fieldValueStr[0]]
+			if ok {
+				return ptr.String(cachedValue), nil
+			}
+			areaID, err := uuid.FromString(fieldValueStr[0])
+			if err != nil {
+				return nil, errs.Wrapf(err, "failed to convert area type value to string for field key: %s", fieldKey)
+			}
+			area, err := app.Areas().Load(ctx, areaID)
+			if err != nil {
+				return nil, errs.Wrapf(err, "failed to retrieve area for field key: %s", fieldKey)
+			}
+			(*uuidStringCache)[fieldValueStr[0]] = area.Name
+			return ptr.String(area.Name), nil
+		case workitem.KindLabel:
+			cachedValue, ok := (*uuidStringCache)[fieldValueStr[0]]
+			if ok {
+				return ptr.String(cachedValue), nil
+			}
+			labelID, err := uuid.FromString(fieldValueStr[0])
+			if err != nil {
+				return nil, errs.Wrapf(err, "failed to convert label type value to string for field key: %s", fieldKey)
+			}
+			label, err := app.Labels().Load(ctx, labelID)
+			if err != nil {
+				return nil, errs.Wrapf(err, "failed to retrieve label for field key: %s", fieldKey)
+			}
+			(*uuidStringCache)[fieldValueStr[0]] = label.Name
+			return ptr.String(label.Name), nil
+		default:
+			// the default case is also used for KindBoardcolumn as resolving the column is not provided by the
+			// factories and the resolved name also has limited use for the exported data.
+			return ptr.String(fieldValueStr[0]), nil
+		}
+	} else if len(fieldValueStr) == 1 {
```

Comment:
Can't this be a single character? What if I set title to `a` ? That would be length=1 string.
If this check is for nil then you should change it to `fieldValue == ""`

Reply:
`fieldValueStr` is not a `string`, it is a `[]string`, so the check is correct :-)

Reply:
The comment there is ambigous, I'll fix this.

Reply:
@michaelkleinhenz I guess not the comment but the variable name is confusing. `fieldValueStr` is a slice of strings and not a string. Maybe call it `fieldValueStrSlice`?

Reply:
@michaelkleinhenz `fieldValueStr` is confusing if the actual variable is `[]string`. Let's rename it to something more appropriate.

Reply:
Fixed 27cc891.